google/cloud/bigquery/_helpers.py

```diff
@@ -172,3 +172,55 @@ def _validate(self, value):
         """
         if value not in self.ALLOWED:
             raise ValueError('Pass one of: %s' ', '.join(self.ALLOWED))
+
+
+class UDFResource(object):
```

Comment:
@tswast the move was part of reducing circular dependencies between `query.py` and `job.py`.